File: internal/infrastructure/secrets/secrets.go
package secrets

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
)

type SecretsManagerClient struct {
	client *secretsmanager.Client
}

func NewSecretsManagerClient(ctx context.Context, cfg aws.Config) (*SecretsManagerClient, error) {
	return &SecretsManagerClient{
		client: secretsmanager.NewFromConfig(cfg),
	}, nil
}

func (s *SecretsManagerClient) Get(ctx context.Context, secretID string) (string, error) {
	input := &secretsmanager.GetSecretValueInput{
		SecretId: aws.String(secretID),
	}
	output, err := s.client.GetSecretValue(ctx, input)
	if err != nil {
		return "", fmt.Errorf("Secrets Manager GetSecretValue failed for %s: %w", secretID, err)
	}
	if output.SecretString == nil {
		return "", fmt.Errorf("secret %s not found or empty", secretID)
	}
	return *output.SecretString, nil
}

File: tracking/src/aws/secrets.rs
use aws_sdk_secretsmanager::Client;

pub struct SecretsManagerClient {
    client: Client,
}

impl SecretsManagerClient {
    pub fn new(config: &aws_config::SdkConfig) -> Self {
        Self {
            client: Client::new(config),
        }
    }

    pub async fn get(
        &self,
        secret_id: &str,
    ) -> Result<String, Box<dyn std::error::Error + Send + Sync>> {
        let output = self
            .client
            .get_secret_value()
            .secret_id(secret_id)
            .send()
            .await?;

        output
            .secret_string
            .ok_or_else(|| format!("Secret {} not found", secret_id).into())
    }

    pub async fn get_optional(&self, secret_id: &str) -> Option<String> {
        self.get(secret_id).await.ok()
    }
}
